### DISCUSSION TITLE
Choppy simulation in version 2.1.2.0

### DISCUSSION THREAD

**Discussion by Hurleyworks** [InteractiveComputerGraphics/SPlisHSPlasH#216] [General]
Howdy,

Thanks for the update!

I just downloaded the latest version 2.1.2.0 and find that the simulation playback is "choppy" compared to version  2.11.6 on my machine.

Windows 11 Pro
Intel(R) Core(TM) i9-12900KF   3.20 GHz  16 cores 64.0 GB RAM
NVIDIA GeForce RTX 3090 24 GB

Here's a video of the emitter demo, first with version 2.1.2.0 and then with 2.1.1.6
https://youtu.be/AEzCEvMHoAs

**Comment by janbender** [comment]
Thanks for reporting this. In the last version I reimplemented the pressure solver and updated some GUI stuff. I assume this could be a GUI problem. Could you please perform the following test with both versions on your maching: 

- Start the simulator in the command line:
SPHSimulator.exe --no-gui --stopAt 3 ..\data\Scenes\Emitter.json

- In the end of the simulation there is an output of timings. Can you compare them to see if the new pressure solver is slower than the old one (when doing a simulation without GUI). 

One problem could be the vsync. You could also try to disable this temporarily in the graphics driver.

**Comment by Hurleyworks** [comment]
Here's the timing outputs that I got with  SPHSimulator.exe --no-gui
--stopAt 3 ..\data\Scenes\Emitter.json

SPlisHSPlasH version: 2.12.0
----------------------------
Average time: neighborhood_search: 0.25088 ms
Average time: precomputeValues: 0.0668737 ms
Average time: computeVolumeAndBoundaryX: 0.137818 ms
Average time: computeDFSPHFactor: 0.0366977 ms
Average time: divergenceSolve: 0.198685 ms
Average time: computeNonPressureForces: 0.057501 ms
Average time: pressureSolve: 0.26624 ms
Average time: emitParticles: 0.039452 ms
Average time: animateParticles: 0.000320812 ms
Average time: SimStep: 1.18596 ms

SPlisHSPlasH version: 2.11.6
------------------------------
Average time: neighborhood_search: 0.400476 ms
Average time: computeVolumeAndBoundaryX: 0.148179 ms
Average time: computeDFSPHFactor: 0.0123474 ms
Average time: divergenceSolve: 0.110607 ms
Average time: computeNonPressureForces: 0.0372581 ms
Average time: pressureSolve: 0.151835 ms
Average time: emitParticles: 0.00876213 ms
Average time: animateParticles: 0.000235304 ms
Average time: SimStep: 0.973975 ms

SPlisHSPlasH version: 2.12.0
----------------------------
Time sum: neighborhood_search: 290.52 ms
Time sum: precomputeValues: 77.4397 ms
Time sum: computeVolumeAndBoundaryX: 159.593 ms
Time sum: computeDFSPHFactor: 42.4959 ms
Time sum: divergenceSolve: 230.078 ms
Time sum: computeNonPressureForces: 66.5862 ms
Time sum: pressureSolve: 308.306 ms
Time sum: emitParticles: 45.6854 ms
Time sum: animateParticles: 0.3715 ms
Time sum: SimStep: 1373.34 ms

SPlisHSPlasH version: 2.11.6
------------------------------
Time sum: neighborhood_search: 467.355 ms
Time sum: computeVolumeAndBoundaryX: 172.924 ms
Time sum: computeDFSPHFactor: 14.4094 ms
Time sum: divergenceSolve: 129.079 ms
Time sum: computeNonPressureForces: 43.4802 ms
Time sum: pressureSolve: 177.191 ms
Time sum: emitParticles: 10.2254 ms
Time sum: animateParticles: 0.2746 ms
Time sum: SimStep: 1136.63 ms

On Wed, Nov 30, 2022 at 2:37 AM Jan Bender ***@***.***> wrote:

> Thanks for reporting this. In the last version I reimplemented the
> pressure solver and updated some GUI stuff. I assume this could be a GUI
> problem. Could you please perform the following test with both versions on
> your maching:
>
>    -
>
>    Start the simulator in the command line:
>    SPHSimulator.exe --no-gui --stopAt 3 ..\data\Scenes\Emitter.json
>    -
>
>    In the end of the simulation there is an output of timings. Can you
>    compare them to see if the new pressure solver is slower than the old one
>    (when doing a simulation without GUI).
>
> One problem could be the vsync. You could also try to disable this
> temporarily in the graphics driver.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/discussions/216#discussioncomment-4271248>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB56PP2UNB4SOVNQNBDC6GDWK372BANCNFSM6AAAAAASO5BEQI>
> .
> You are receiving this because you authored the thread.Message ID:
> <InteractiveComputerGraphics/SPlisHSPlasH/repo-discussions/216/comments/4271248
> @github.com>
>


-- 
-Steve
http://berserko.hurleyworks.com/ <http://www.hurleyworks.com>

**Comment by janbender** [comment]
Thanks for the numbers. I will test this on my system. 
Did you try to turn off the vsync? On my system this is an issue. Alternatively you can build the simulator with the modified file I attached. Just exchange the corresponding file in GUI/OpenGL

[MiniGL.zip](https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/10122670/MiniGL.zip)

**Comment by Hurleyworks** [comment]
>
> Alternatively you can build the simulator with the modified file I
> attached. Just exchange the corresponding file in GUI/OpenGL


That fixed it.  All the sims are running much smoother here now. Thanks!

On Wed, Nov 30, 2022 at 6:26 AM Jan Bender ***@***.***> wrote:

> Thanks for the numbers. I will test this on my system.
> Did you try to turn off the vsync? On my system this is an issue.
> Alternatively you can build the simulator with the modified file I
> attached. Just exchange the corresponding file in GUI/OpenGL
>
> MiniGL.zip
> <https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/files/10122670/MiniGL.zip>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/discussions/216#discussioncomment-4272868>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AB56PP7YAWLCVBN7BF5X4QTWK42WDANCNFSM6AAAAAASO5BEQI>
> .
> You are receiving this because you authored the thread.Message ID:
> <InteractiveComputerGraphics/SPlisHSPlasH/repo-discussions/216/comments/4272868
> @github.com>
>


-- 
-Steve
http://berserko.hurleyworks.com/ <http://www.hurleyworks.com>
